thesis/03-implementation.tex

```diff
@@ -83,23 +83,491 @@ \subsubsection{Backend}
+\end{figure}
+
+
+\subsubsection{Community}
+
+The community component is written in \tech{Python} using the \tech{Django} framework and therefore uses the \tech{unittest} module that is built-in to the Python standard library. All of the public-facing endpoints are fully covered with integration tests verifying that authentication, authorization, and request body handling are properly implemented. The line test coverage is 96\%.
+
+\medskip
+Code style abides by the \tech{PEP8} standard.
+
+\medskip
+To ensure build reproducibility, the community component is enclosed in a Docker image, which is automatically deployed to a production environment. 
+
+\begin{figure}[H]
+  \caption{Pipeline - Community backend component.}
+  \centering
+    \includegraphics[width=\textwidth]{assets/3-community-pipeline.png}
+\end{figure}
+
+\subsection{Deployment}
+
+Origuide uses Docker images for packaging components. Images are pushed to the \tech{Docker Hub} repository where they are publicly available.
+
+\medskip
+The deployment configuration is specified using \tech{docker-compose} files and is environment agnostic due to having environment configurations externalized to \texttt{.env} files. This makes \tech{Origuide}'s backend deployable using a single command.
+
+\begin{figure}[H]
+	\caption{Example configuration of a local environment in \textit{.env} file}
+	\lstinputlisting{assets/listings/3-8-env-file}
+\end{figure}
+
+\begin{figure}[H]
+	\caption{Origuide backend can be deployed using a single command}
+	\begin{lstlisting}
+	docker-compose up -d
+	\end{lstlisting}
+\end{figure}
+
+The backend production environment is running on servers provided by \tech{Digital Ocean}.
+
+\medskip
+The frontend is built and deployed on \tech{Netlify} with no additional configuration requirements. This configuration-less nature of the deployment allows switching hosting providers easily should a need arise.
+
+
+
+
 % Instrukcja postawienia środowiska deweloperskiego - jeśli potrzeba wypełniacza %
 
-\subsection{Quality assurance}
-% Quality Assurance: czy mamy testy jednostkowe? Czy mamy inne testy automatyczne? Jakich bibliotek używacie? %
 
 \subsection{Problems encountered}
 % Z jakimi problemami technicznymi sie borykaliście? Jeśli nie opisaliście ich w dokumentacji procesowej. %
+% Collissions
+\subsubsection{Collision detection and handling}
+\subsubsection{Damping force}
+\subsubsection{Paper stiffness}
+\subsubsection{Unnatural guide encoding}
+
+\subsubsection{Mesh rendering hazard}
+
+When folding origami models, two faces lying flat on each other are a common occurrence.
+
+The Solving algorithm is not sophisticated enough, to have a sense of sheet thickness. Therefore it places the folded faces close to each other, on the verge of numerical error. The rendering engine cannot distinguish which face is on top and which one is below, resulting in face permeation and color leakage demonstrated below.
```

Comment:
not sophisticated enough, to have -> no comma
Therefore it -> Therefore, it (with comma)
I still think permeation is not a right word. Sounds fancy, admitted, but it seems too specific to apply to this problem.

Kind of being picky. Color leakage, hmm. It's not leaking. Color mixture? Mixing of colors? Color confusion?

Reply:
I think the color is leaking.

I agree with the permeation. I think it fits better here than on the "flipped" case (where it didn't). Removed the faces permeation because it's mentioned at the beginning of the sentence.

Changed it to `resulting in face color mixture`. Maybe we also need an article before face, not sure.